msteams-platform/concepts/bots/bots-files.md

```diff
@@ -0,0 +1,215 @@
+    "name": "file_example.txt",
+    "content": {
+      "uniqueId": "1150D938-8870-4044-9F2C-5BBDEBA70C8C",
+      "fileType": "txt",
+    }
+  }]
+}
+```
+
+The following table describes the content properties of the attachment:
+
+| Property | Purpose |
+| --- | --- |
+| `uniqueId` | OneDrive/SharePoint drive item ID. |
+| `fileType` | File type, such as pdf or docx. |
+
+## Basic example in C#
+The following sample shows how you can handle file uploads and send file consent requests in your bot's dialog.
+
+```csharp
+
+// This sample dialog shows two simple flows:
+// 1) A silly example of receiving a file from the user, processing the key elements,
+//    and then constructing the attachment and sending it back.
+// 2) Creating a new file consent card requesting user permission to upload a file.
+private async Task MessageReceivedAsync(IDialogContext context, IAwaitable<object> result)
+{
+    var replyMessage = context.MakeMessage();
+    Attachment returnCard;
+
+    var message = await result as Activity;
+
+    // Check to see if the user is sending the bot a file.
+    if (message.Attachments != null && message.Attachments.Any())
+    {
+        var attachment = message.Attachments.First();
+
+        if (attachment.ContentType == FileDownloadInfo.ContentType)
+        {
+            FileDownloadInfo downloadInfo = (attachment.Content as JObject).ToObject<FileDownloadInfo>();
+            if (downloadInfo != null)
+            {
+                returnCard = CreateFileInfoAttachment(downloadInfo, attachment.Name, attachment.ContentUrl);
+                replyMessage.Attachments.Add(returnCard);
+            }
+        }
+    }
+    else
+    {
+        // Illustrates creating a file consent card.
+        returnCard = CreateFileConsentAttachment();
+        replyMessage.Attachments.Add(returnCard);
+    }
+    await context.PostAsync(replyMessage);
+}
+
+
+private static Attachment CreateFileInfoAttachment(FileDownloadInfo downloadInfo, string name, string contentUrl)
+{
+    FileInfoCard card = new FileInfoCard()
+    {
+        FileType = downloadInfo.FileType,
+        UniqueId = downloadInfo.UniqueId
+    };
+
+    Attachment att = card.ToAttachment();
+    att.ContentUrl = contentUrl;
+    att.Name = name;
+
+    return att;
+}
+
+private static Attachment CreateFileConsentAttachment()
+{
+    JObject acceptContext = new JObject();
+    // Fill in any additional context to be sent back when the user accepts the file.
+
+    JObject declineContext = new JObject();
+    // Fill in any additional context to be sent back when the user declines the file.
+
+    FileConsentCard card = new FileConsentCard()
+    {
+        AcceptContext = acceptContext,
+        DeclineContext = declineContext,
+        SizeInBytes = 102635,
+        Description = "File description"
+    };
+
+    Attachment att = card.ToAttachment();
+    att.Name = "Example file";
```

Comment:
Why can't Name be a property of FileConsentCard, and ToAttachment() handles this?

Reply:
I've asked Ramjot to make the change to the SDK, but right now you have to set it as a separate property because it's on the attachment object and not the content object.